### ISSUE TITLE
Warning message when trying to use parallel processing

### ISSUE THREAD

**Issue by williash23** [prioritizr/prioritizr#100]
Hello,

Trying to set up a problem using more than one thread. Here is my setup:

`p1 <- problem(x = pu_in, features = feat_in, cost_column = "area_h") %>%
		add_min_set_objective() %>%
		add_relative_targets(tar) %>%
		add_locked_in_constraints(locked_in) %>%
		add_binary_decisions() %>%
		add_gurobi_solver()`

If I run this problem and solve without parallel processing, I have no issues (but it takes a really long time). In an effort to make this problem run more quickly, I am trying to use multiple threads and so run this before setting up the problem or solving:

`set_number_of_threads(3)`

However, I am getting this warning message:

![image](https://user-images.githubusercontent.com/11653020/46305010-62883180-c56d-11e8-8348-e626e89a2e58.png)

I'm not sure if this is causing/can cause an actual problem, but figured I would try to see if anyone has run into this message before. 

Thanks!

**Comment by williash23** [comment]
Thanks so much for the response and the explanation of parallel processing before and during the optimization. 

Here is my set up:

**Planning units** 
8,073 units all with nearly the same cost (area_h is between  499 and 500 for all planning units)
![image](https://user-images.githubusercontent.com/11653020/46428535-72387f00-c701-11e8-898b-ee0579312a49.png)

**Conservation features**
77 rasters in a stack, with cell values that range from 0 to 1.
![image](https://user-images.githubusercontent.com/11653020/46428558-8a100300-c701-11e8-9c51-b58eb0ee89a4.png)

**And the problem set up**
![image](https://user-images.githubusercontent.com/11653020/46428687-e2470500-c701-11e8-9349-5497f755b8a6.png)


You are right - solving the  problem takes less than a minute. Setting up the problem is the time consuming step - for this instance it took 37 minutes. Will using `set_number_of_threads()` help shorten the time problem set up takes?

**Comment by jeffreyhanson** [comment]
Thanks for sharing this info. Yeah, using `set_number_of_threads` should help. Also, playing around with the maximum amount of memory the _raster_ R package can use also helps too (e.g. `raster::rasterOptions(maxmemory = 1e+11, chunksize = 1e+10)` , the exact numbers will depend on your RAM). If possible, I'd recommend using raster planning units (with pixel values corresponding to cost), as the one of the most computationally intensive tasks in the data prep stage is overlaying the polygons with the raster stack (this will probably have the biggest effect on reducing processing time).

Also, the minimum set objective doesn't use weights (there should be some warning printed about this - let me know if there isn't?). It will always deliver solutions that meet the targets (though the efficiency of a solution depends on how far from optimality the solution is). So, if different features require different levels of representation, then you should try setting different targets with `add_relative_targets`. The weights are used for objectives (e.g. `add_max_feature_objective`, `add_max_cover_objective`) where the overall aim is to maximise the protected biodiversity subject to a budget. For these objectives, you can specify weights to indicate the relative importance for protecting one species over another.

**Comment by williash23** [comment]
Sorry I took so long to respond - but just wanted to say thanks for the further direction. I did end up converting my planning unit layer to raster format. As you said, this _drastically_ reduced the processing time to set up the problem. 

I also did fix the situation of using weights with the minimum set objective (I adjusted with relative targets instead). I was not getting an error from the problem set up before when I was trying to use weights.

Thanks again for your insight!
